mne/beamformer/_dics.py

```diff
@@ -464,16 +469,18 @@ def apply_dics_csd(csd, filters, verbose=None):
     frequencies = [np.mean(dfreq) for dfreq in csd.frequencies]
     n_freqs = len(frequencies)
 
-    source_power = np.zeros((n_sources, len(csd.frequencies)))
+    source_power = np.zeros(
+        (n_sources * (3 if filters['pick_ori'] == 'vector' else 1),
+         len(csd.frequencies)))
```

Comment:
This is not necessary. `apply_dics_csd` computes power, which has no orientation. The actual power computation happens below in the `_compute_power` function. If you feed a DICS beamformer object with `pick_ori='vector'` to this function, it just pools power across all orientations. Hence, the result is always just 1 scalar value per vertex.

Reply:
Sounds reasonable and I'm not at all an expert in this but just for clarification, could you just return power in each (x, y, z) direction computed separately and independently? Not sure if that even makes sense as a question but that was my thought process in how one might use the vector source estimate. That might, instead of pooling across all orientations, separate them out. Does that make any sense?

Reply:
You certainly could, but it would not be terribly useful. The xyz directions are mostly arbitrary. They could be the global xyz, or z can be aligned with the surface normal and x and y being arbitrary directions. What makes these directions useful is that they are perpendicular hence forming a coordinate system. Combine the activity in each direction and you have a vector pointing in the orientation of the source activity. This is what happens when you call `VectorSourceEstimate.plot()`: you get pretty arrows showing the directions of the vectors.

Unfortunately, this doesn't work when you have power, the signal squared, in each direction. It's no longer linear, hence you can't combine the xyz directions into a vector anymore. One obvious way in which this fails is that squared values are always positive, hence the vector would be restricted to a 90 degree cone, regardless of the true source orientation, otherwise at least one of the directions would have to be negative.

So, simply combining the directions like we would with "normal" time course source activity doesn't work. There is a not-so-simple way to combine them though, namely by using the largest eigenvalue of the xyz source CSD matrix (applying the DICS filters to a CSD will actually give you a full source CSD, from which we now just take the diagonal to compute power, but you could do more with it). However, this would be a bigger project. This is something that is not typically done, so we would need to verify the method on some datasets to make sure we got it right, and at that point it's probably only worth it if we had some researchers that have an actual direct need to have this functionality.

Reply:
Ah thanks for explaining, that makes sense! I can write a little note explaining that vector DICS filters will return scalar source estimates just so it's in the documentation.